urld58fd5e7a9_longrm_weibo_robot_tgz-pJ8-weibo4j_org_json_TestJ8/src/weibo4j/RetweetDetails.java

```diff
@@ -0,0 +1,176 @@
+*/
+package weibo4j;
+
+import org.w3c.dom.Element;
+import org.w3c.dom.Document;
+import org.w3c.dom.NodeList;
+
+import weibo4j.http.Response;
+import weibo4j.org.json.JSONArray;
+import weibo4j.org.json.JSONException;
+import weibo4j.org.json.JSONObject;
+
+import java.util.Date;
+import java.util.List;
+import java.util.ArrayList;
+
+/**
+ * A data class representing one single retweet details.
+ *
+ * @author Yusuke Yamamoto - yusuke at mac.com
+ * @since Weibo4J 1.2.0
+ */
+@Deprecated
+public class RetweetDetails extends WeiboResponse implements
+        java.io.Serializable {
+    private long retweetId;
+    private Date retweetedAt;
+    private User retweetingUser;
+    static final long serialVersionUID = 1957982268696560598L;
+    
+    /*package*/RetweetDetails(Response res, Weibo weibo) throws WeiboException {
+        super(res);
+        Element elem = res.asDocument().getDocumentElement();
+        init(res, elem, weibo);
+    }
+    
+    RetweetDetails(JSONObject json) throws WeiboException {
+        super();
+        init(json);
+    }
+    
+    
+
+    private void init(JSONObject json) throws WeiboException{
+    	try {
+    		retweetId = json.getInt("retweetId");
+    		retweetedAt = parseDate(json.getString("retweetedAt"), "EEE MMM dd HH:mm:ss z yyyy");
+    		retweetingUser=new User(json.getJSONObject("retweetingUser"));
+            
+        } catch (JSONException jsone) {
+            throw new WeiboException(jsone.getMessage() + ":" + json.toString(), jsone);
+        }
+	}
+
+	/*package*/RetweetDetails(Response res, Element elem, Weibo weibo) throws
+            WeiboException {
+        super(res);
+        init(res, elem, weibo);
+    }
+
+    private void init(Response res, Element elem, Weibo weibo) throws
+            WeiboException {
+        ensureRootNodeNameIs("retweet_details", elem);
+        retweetId = getChildLong("retweet_id", elem);
+        retweetedAt = getChildDate("retweeted_at", elem);
+        retweetingUser = new User(res, (Element) elem.getElementsByTagName("retweeting_user").item(0)
+                , weibo);
+    }
+
+    public long getRetweetId() {
+        return retweetId;
+    }
+
+    public Date getRetweetedAt() {
+        return retweetedAt;
+    }
+
+    public User getRetweetingUser() {
+        return retweetingUser;
+    }
+    /*modify by sycheng add json*/
+    /*package*/
+    static List<RetweetDetails> createRetweetDetails(Response res) throws WeiboException {
+    	try {
+            JSONArray list = res.asJSONArray();
+            int size = list.length();
+            List<RetweetDetails> retweets = new ArrayList<RetweetDetails>(size);
+            for (int i = 0; i < size; i++) {
+            	retweets.add(new RetweetDetails(list.getJSONObject(i)));
+            }
+            return retweets;
+        } catch (JSONException jsone) {
+            throw new WeiboException(jsone);
+        } catch (WeiboException te) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An exception is caught and re-thrown within the catch clause without any other operations being performed on the exception, making the catch clause unnecessary.

**Fix**: Perform additional operations on the exception that has been caught. You can wrap a standard exception into a custom exception to add additional information or special handling. When you do that, make sure to set the original exception as the cause. Otherwise, you will lose the stack trace and message of the original exception, which will make it difficult to analyze the event that caused the exception.  
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)